### PR TITLE
migration: sketch out working migration

### PR DESCRIPTION
**Pull request by liam-fitzgerald**
Depends on the `lf/gladio-2` branch living in landscape

### To run the migration
`-landscape!migrate`

### Notes
Shaping up pretty nicely, addressed most issues uncovered in the most recent testate run

### CODE REVIEW

File: desk/lib/chat-graph.hoon [hunk 1]
```diff
@@ -28,21 +28,63 @@
...
-  ^-  inline:c
+  ^-  (list inline:c)
   ?-  -.con
-    %text       text.con
-    %mention    `@t`(scot %p ship.con)
-    %url        [%link [. .]:url.con]
-    %code       [%inline-code expression.con]
-    %reference  'elided reference' :: TODO fix?
+    %text       (fall (rush text.con apex:parse) ~[text.con])
+    %mention    ~[`@t`(scot %p ship.con)]
```
Comment (arthyn) [inline desk/lib/chat-graph.hoon:48]: I think we can use the actual mention type now

File: desk/app/chat.hoon [hunk 2]
```diff
@@ -156,7 +156,7 @@
     |=  =chat:zero
     %*  .  *chat:one
       remark  remark.chat
-      log     log.chat
+      ::  log     *log.chat
       perm    perm.chat
       pact    pact.chat
```
Comment (arthyn) [inline desk/app/chat.hoon:161]: I just ripped out all the state migrations in jake's branch since we're nuking anyway

**Review by arthyn — COMMENTED** [review]
I think the flow of this makes sense. Left a few questions. One general question is are we going to do all the desk/agent renames with a slew of pokes from %groups or how do we encode that in the OTA?

File: desk/app/groups.hoon [hunk 3]
```diff
@@ -374,7 +407,7 @@
     ?+  module.config.metadatum  ~
       %chat  `%chat
       %link  `%heap
-      %publish   ~  :: TODO
+      %publish   `%diary  :: TODO
```
[the remark this answers is not in the document]
Comment (liam-fitzgerald) [reply desk/app/groups.hoon:410]: nope

**Review by arthyn — APPROVED** [review]
we've pored over this enough, lgtm

**Pull request merged**